### PR TITLE
[REVIEW] fix dangling exec_policy pointer and invalid num_ring argument

### PR DESCRIPTION
**Pull request by zhangjianting**
(1) fix invalid pointer bug in `auto exec_policy=rmm::exec_policy(stream)->on(stream)` by declaring `rmm::exec_policy(stream)` first and then using the variable with `->on(stream)`. 
(2) fix wrong vertex positions in polygon_soa_reader and shapefile_soa_reader. The bug affects multi-polygons with multiple rings only.

### CODE REVIEW

File: cpp/src/io/shp/polygon_shapefile_reader.cu [hunk 1]
```diff
@@ -199,14 +199,15 @@ namespace cuspatial
...
+        auto exec_policy = rmm::exec_policy(stream);    
 
         int32_t* temp{nullptr};
         RMM_TRY( RMM_ALLOC(&temp, pm.num_feature * sizeof(int32_t), stream) );
         CUDA_TRY( cudaMemcpyAsync(temp, pm.feature_length,
                               pm.num_feature * sizeof(int32_t),
                               cudaMemcpyHostToDevice, stream) );
         //prefix-sum: len to pos
-        thrust::inclusive_scan(rmm::exec_policy(stream)->on(stream), temp, temp + pm.num_feature, temp);
+        thrust::inclusive_scan(exec_policy->on(stream), temp, temp + pm.num_feature, temp);
```
Comment (jrhemstad) [inline cpp/src/io/shp/polygon_shapefile_reader.cu]: The old usage of `rmm::exec_policy(stream)->on(stream)` is just fine.

What's problematic is when you do this:

```
auto exec = rmm::exec_policy(stream)->on(stream);
```

Comment (zhangjianting) [reply cpp/src/io/shp/polygon_shapefile_reader.cu]: understood. I wanted to simplify code, especially when exec_policy is used in quite some places.

Comment (zhangjianting) [reply cpp/src/io/shp/polygon_shapefile_reader.cu]: I will change the description to reflex the bug more precisely.

**Comment by harrism** [comment]
If there was a crash bug in 0.13 perhaps this should go into branch-0.13 so we don't risk having to do a hot fix? Need to decide now because code freeze starts soon.

**Comment by zhangjianting** [comment]
> If there was a crash bug in 0.13 perhaps this should go into branch-0.13 so we don't risk having to do a hot fix? Need to decide now because code freeze starts soon.

It would be great to get the bug fix into 0.13. Can I re-target to 0.13?

File: CHANGELOG.md [hunk 2]
```diff
@@ -5,6 +5,7 @@
 ## Improvements
 
 ## Bug Fixes
+- PR #141 fix invalid pointer bug in 'auto exec_policy=rmm::exec_policy(stream)->on(stream)' and wrong num_ring for multi-polygons
```
Comment (harrism) [inline CHANGELOG.md]: Shorter changelog message, please.
```suggestion
- PR #141 fix dangling exec_policy pointer and invalid num_ring argument.
```

**Pull request merged**